src/terminal/terminal.c

```diff
@@ -1590,17 +1590,129 @@ static int __guac_terminal_send_key(guac_terminal* term, int keysym, int pressed
+            }
+
+            /* Shift + Alt + Function keys. */
+            if(term->mod_shift && term->mod_alt){
+                if (keysym == 0xFFBE || keysym == 0xFF91) return guac_terminal_send_string(term, "\x1B[1;4P"); /* F1  */
+                if (keysym == 0xFFBF || keysym == 0xFF92) return guac_terminal_send_string(term, "\x1B[1;4Q"); /* F2  */
+                if (keysym == 0xFFC0 || keysym == 0xFF93) return guac_terminal_send_string(term, "\x1B[1;4R"); /* F3  */
+                if (keysym == 0xFFC1 || keysym == 0xFF94) return guac_terminal_send_string(term, "\x1B[1;4S"); /* F4  */
+                if (keysym == 0xFFC2) return guac_terminal_send_string(term, "\x1B[15;4~"); /* F5  */
+                if (keysym == 0xFFC3) return guac_terminal_send_string(term, "\x1B[17;4~"); /* F6  */
+                if (keysym == 0xFFC4) return guac_terminal_send_string(term, "\x1B[18;4~"); /* F7  */
+                if (keysym == 0xFFC5) return guac_terminal_send_string(term, "\x1B[19;4~"); /* F8  */
+                if (keysym == 0xFFC6) return guac_terminal_send_string(term, "\x1B[20;4~"); /* F9  */
+                if (keysym == 0xFFC7) return guac_terminal_send_string(term, "\x1B[21;4~"); /* F10 */
+                if (keysym == 0xFFC8) return guac_terminal_send_string(term, "\x1B[23;4~"); /* F11 */
+                if (keysym == 0xFFC9) return guac_terminal_send_string(term, "\x1B[24;4~"); /* F12 */
+            }
+
+            /* Ctrl+ Function keys. */
+            if (term->mod_ctrl) { 
+                if (keysym == 0xFFBE || keysym == 0xFF91) return guac_terminal_send_string(term, "\x1B[1;5P"); /* F1  */
+                if (keysym == 0xFFBF || keysym == 0xFF92) return guac_terminal_send_string(term, "\x1B[1;5Q"); /* F2  */
+                if (keysym == 0xFFC0 || keysym == 0xFF93) return guac_terminal_send_string(term, "\x1B[1;5R"); /* F3  */
+                if (keysym == 0xFFC1 || keysym == 0xFF94) return guac_terminal_send_string(term, "\x1B[1;5S"); /* F4  */
+                if (keysym == 0xFFC2) return guac_terminal_send_string(term, "\x1B[15;5~"); /* F5  */
+                if (keysym == 0xFFC3) return guac_terminal_send_string(term, "\x1B[17;5~"); /* F6  */
+                if (keysym == 0xFFC4) return guac_terminal_send_string(term, "\x1B[18;5~"); /* F7  */
+                if (keysym == 0xFFC5) return guac_terminal_send_string(term, "\x1B[19;5~"); /* F8  */
+                if (keysym == 0xFFC6) return guac_terminal_send_string(term, "\x1B[20;5~"); /* F9  */
+                if (keysym == 0xFFC7) return guac_terminal_send_string(term, "\x1B[21;5~"); /* F10 */
+                if (keysym == 0xFFC8) return guac_terminal_send_string(term, "\x1B[23;5~"); /* F11 */
+                if (keysym == 0xFFC9) return guac_terminal_send_string(term, "\x1B[24;5~"); /* F12 */
+            }
+
+            /* Shift + Ctrl + Function keys. */
+            if(term->mod_shift && term->mod_ctrl){
```

Comment:
Same here:

* Please follow established style.
* This will never actually be reached as earlier `if` tests will have already been satisfied and the function will already have exited.